dill/_shims.py

```diff
@@ -264,3 +264,49 @@ def _delattr(cell, name):
 
 _setattr = Getattr(_dill, '_setattr', setattr)
 _delattr = Getattr(_dill, '_delattr', delattr)
+
+# Structural Pattern Matching for reduce tuples.
+@move_to(_dill)
+class match:
```

Comment:
Why is this defined in `shims`?  Shims are intended to enable old stored pickles to be unserialized when a create or save function has changed.  Can you also update the PR with regard to changes in `shims`?

Reply:
`shims` makes a lot of sense now... I thought that it was for compatibility with old _and supported_ (but older) python/dill versions. OK, I'll move it to `_dill.py`.